cadasta/gui/tools/wizard/step_project_creation02.py

```diff
@@ -0,0 +1,127 @@
+
+    def __init__(self, parent=None):
+        """Constructor
+
+        :param parent: parent - widget to use as parent.
+        :type parent: QWidget
+        """
+        super(StepProjectCreation2, self).__init__(parent)
+        self.layer = None
+        self.layer_attributes = None
+
+    def attributes(self):
+        """Returns data from the layer
+
+        :returns: attributes layer selected
+        :rtype: list of dict or none
+        """
+        cadasta_field = {
+            'location_attribute': self.location_attribute_box.currentText(),
+            'location_type': self.location_type_box.currentText(),
+            'party_name': self.party_name_box.currentText(),
+            'party_type': self.party_type_box.currentText(),
+            'party_attribute': self.party_attribute_box.currentText(),
+            'relationship_type': self.relationship_type_box.currentText(),
+            'relationship_attribute':
+                self.relationship_attribute_box.currentText(),
+
+        }
+
+        cadasta_data = []
+
+        for layer in self.layer_attributes:
+            for key, value in cadasta_field.iteritems():
+                if value in layer:
+                    cadasta_data.append(
+                        {key: layer[value] if layer[value] else ''}
+                    )
+
+        return cadasta_data
+
+    def set_widgets(self):
+        """Set all widgets on the tab."""
+        if not self.layer or self.layer != self.parent.layer:
+            self.layer = self.parent.layer
+            self.set_attribute_box()
+
+    def set_attributes_box(self):
+        """Set all attribute box widgets."""
+        field_names = [field.name() for field in self.layer.pendingFields()]
+
+        self.layer_attributes = []
+
+        for elem in self.layer.getFeatures():
+            self.layer_attributes.append(
+                (dict(zip(field_names, elem.attributes())))
+            )
+
+        if 'id' in field_names:
+            field_names.remove(u'id')
+
+        field_names.insert(0, ' ')
+
+        self.location_attribute_box.clear()
+        self.location_attribute_box.addItems(field_names)
+
+        self.location_type_box.clear()
+        self.location_type_box.addItems(field_names)
+
+        self.party_name_box.clear()
+        self.party_name_box.addItems(field_names)
+
+        self.relationship_type_box.clear()
+        self.relationship_type_box.addItems(field_names)
+
+        self.party_type_box.clear()
+        self.party_type_box.addItems(field_names)
+
+        self.party_attribute_box.clear()
+        self.party_attribute_box.addItems(field_names)
+
+        self.relationship_attribute_box.clear()
+        self.relationship_attribute_box.addItems(field_names)
+
+    def validate_step(self):
+        """Check if the step is valid.
+
+        :returns: Tuple of validation status and error message if any
+        :rtype: ( bool, str )
+        """
+        return True, ''
```

Comment:
Only for this step, because all of the input is optional so you can skip this step without filling anything.